control_panel_api/aws.py

```diff
@@ -1,106 +1,109 @@
 import json
 
-from django.conf import settings
-
-
-aws_api_client = settings.AWS_API_CLIENT_HANDLER
-
-
-def create_bucket(name, region, acl='private'):
-    aws_api_client('s3').create_bucket(
-        Bucket=name,
-        ACL=acl,
-        CreateBucketConfiguration={'LocationConstraint': region},
-    )
-
-
-def put_bucket_logging(name, target_bucket, target_prefix):
-    aws_api_client('s3').put_bucket_logging(
-        Bucket=name,
-        BucketLoggingStatus={
-            'LoggingEnabled': {
-                'TargetBucket': target_bucket,
-                'TargetPrefix': target_prefix
-            }
-        }
-    )
+import boto3
 
+from django.conf import settings
 
-def create_policy(name, policy_document):
-    aws_api_client("iam").create_policy(
-        PolicyName=name,
-        PolicyDocument=json.dumps(policy_document)
-    )
 
+class AWSClient(object):
```

Comment:
I like the refactor to use a client object.

I am somewhat ambivalent towards the usage of feature flags due to the peppering of `if` statements in client objects or libraries.  Have you thought about using polymorphism at runtime to replace with `AWS|HelmDummyClient` implementations which would keep the client code interface the same and cleaner than feature flags?  e.g https://docs.djangoproject.com/en/1.11/topics/cache/#dummy-caching-for-development

Reply:
Yep, that does look like a good approach for this case. Feature flags are desirable for other cases, however. I think I will refactor to this approach for this case later, because I don't want Clive to have to wait.

Reply:
👍